src/python/pants/backend/nfpm/fields/all.py

```diff
@@ -0,0 +1,177 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+from enum import Enum
+from typing import ClassVar
+
+from pants.core.goals.package import OutputPathField
+from pants.engine.target import Dependencies, StringField
+from pants.util.docutil import bin_name
+from pants.util.strutil import help_text
+
+
+class NfpmPackageNameField(StringField):
+    nfpm_alias = "name"
+    alias: ClassVar[str] = "package_name"
+    required = True
+    value: str
+    help = help_text(
+        """
+        The package name.
+        """
+    )
+
+
+class NfpmDependencies(Dependencies):
+    nfpm_alias = ""  # doesn't map directly to a nfpm.yaml field
+
+
+class NfpmArch(Enum):
```

Comment:
probably delete as it is not used for anything but a default value.

Reply:
I added it to the field doc string so the valid values are readily available in --help output. So, I'm keeping this one. GoOS on the other hand is unlikely to use anything but the default. Drop/ignore GoOS enum.